### PR TITLE
Added ability to generate GUI plugins to plugin_template.

### PR DESCRIPTION
**Pull request by Pik-9**
The plugin_template script can generate a GUI plugin now. It has an own template for this.

### CODE REVIEW

File: plugins/plugin_template/__gui_project__/manifest.xml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<package>
+    <description brief="__project__">
+      __description__
+   </description>
+	<maintainer>Matthias Goldhoorn/matthias@goldhoorn.eu</maintainer>
```
Comment (goldhoorn) [inline plugins/plugin_template/__gui_project__/manifest.xml]: I don't want to be the default maintainer

File: plugins/plugin_template/__gui_project__/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+set(PROJECT_VERSION 1.0)
+set(PROJECT_DESCRIPTION "__description__")
+cmake_minimum_required(VERSION 2.8.11)
+include(FindPkgConfig)
+
+find_package(mars)
+mars_defaults()
+define_module_info()
+
+find_package(Qt5Widgets REQUIRED)
```
Comment (goldhoorn) [inline plugins/plugin_template/__gui_project__/CMakeLists.txt]: Please use the same method for finding packages than the other mars compoents to make sure they are consistent even qt5 is not used. (which is currently the case in most setups)

See: https://github.com/rock-simulation/mars/blob/master/gui/CMakeLists.txt

File: plugins/plugin_template/__gui_project__/src/__project___MainWin.h [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+#ifndef __headerDef__MW
+#define __headerDef__MW
+
+#include <QWidget>
+#include <QGridLayout>
+#include <QLabel>
+#include <QPushButton>
+
+class __classname___MainWin : public QWidget
```
Comment (goldhoorn) [inline plugins/plugin_template/__gui_project__/src/__project___MainWin.h]: This is published a dependancy to QT therefore it needs to be added to the deps on pkg-config (linkwise)

File: plugins/plugin_template/__gui_project__/__project__.pc.in [hunk 4]
```diff
@@ -0,0 +1,10 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=@CMAKE_INSTALL_PREFIX@
+libdir=${prefix}/lib
+includedir=${prefix}/include
+
+Name: @PROJECT_NAME@
+Description: @PROJECT_DESCRIPTION@
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -l@PROJECT_NAME@
+Cflags: -I${includedir} -l${@QT_CORE_LIBRARY@} -l${@QT_GUI_LIBRARY@}
```
Comment (goldhoorn) [inline plugins/plugin_template/__gui_project__/__project__.pc.in]: better add this to libs instead to cflags

File: plugins/plugin_template/__gui_project__/manifest.xml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+<package>
+    <description brief="__project__">
+        __description__
+     </description>
+    <maintainer>__author__/__email__</maintainer>
+    <author>__author__/__email__</author>
+    <depend package="simulation/mars/scripts/cmake" />
+    <depend package="simulation/lib_manager" />
+    <depend package="simulation/mars/common/data_broker" />
+    <depend package="simulation/mars/interfaces" />
```
Comment (goldhoorn) [inline plugins/plugin_template/__gui_project__/manifest.xml]: I assume this also depends on mars_gui? isn't it?

**Comment by goldhoorn** [comment]
gihub doesn't notify if you push something new..
please ping here if you want a review.

P.S. i prefer to squash your commits into one during review phase..

**Comment by Pik-9** [comment]
Okay, fixed it.
Should I try to squash the commits or do you want to do that?

**Comment by goldhoorn** [comment]
i prefer a squash what do you think @kavonszadkowski , otherwise ack for merge for me

**Pull request merged**